mifospay/src/main/java/org/mifos/mobilewallet/mifospay/data/firebase/api/services/MifosPayInstanceIDService.java

```diff
@@ -32,9 +32,12 @@
 public class MifosPayInstanceIDService extends FirebaseInstanceIdService {
 
     private static final String TAG = "MyFirebaseIIDService";
+    private final PreferencesHelper mPreferencesHelper;
 
     @Inject
-    PreferencesHelper mPreferencesHelper;
+    public MifosPayInstanceIDService(PreferencesHelper mPreferencesHelper) {
```

Comment:
`FirebaseInstanceIdService` needs a zero args constructor so you cant provide a constructor that has `PreferencesHelper` argument. (It will only call the zero arg constructor, will also be crashing in your latest change)

Instead do it like this - 
```
    @Inject
    PreferencesHelper mPreferencesHelper;

    public MifosPayInstanceIDService() {
        MifosPayApp.get(MifosPayApp.getContext()).component().inject(this);
    }
```

And add this method in `ApplicationComponent`
```
    void inject(MifosPayInstanceIDService mifosPayInstanceIDService);
```